pyanaconda/kickstart.py

```diff
@@ -2001,16 +2004,45 @@ def __init__(self, *args, **kwargs):
         self.enabled = kwargs.pop("enabled", True)
         self.repo_id = kwargs.pop("repo_id", None)
         self.treeinfo_origin = kwargs.pop("treeinfo_origin", False)
+        self.partition = kwargs.pop("partition", None)
+        self.iso_path = kwargs.pop("iso_path", None)
+
+        self.mount_dir_suffix = kwargs.pop("mount_dir_suffix", None)
 
         super().__init__(*args, **kwargs)
 
+    @classmethod
+    def create_copy(cls, other):
+        return RepoData(name=other.name,
+                        baseurl=other.baseurl,
+                        mirrorlist=other.mirrorlist,
+                        metalink=other.metalink,
+                        proxy=other.proxy,
+                        enabled=other.enabled,
+                        treeinfo_origin=other.treeinfo_origin,
+                        partition=other.partition,
+                        iso_path=other.iso_path,
+                        mount_dir_suffix=other.mount_dir_suffix)
+
+    def generate_mount_dir(self):
+        """Generate persistent mount directory suffix
+
+        This is valid only for HD repositories
+        """
+        if self.is_harddrive_based() and self.mount_dir_suffix is None:
+            self.mount_dir_suffix = "addition_" + str(self.__mount_counter)
+            self.__class__.__mount_counter += 1
```

Comment:
It might look better to have a class method `generate_mount_dir_suffix`, that could handle the mount counter.

Reply:
Good idea, thanks!